dyno-core/src/main/java/com/netflix/dyno/connectionpool/Host.java

```diff
@@ -122,13 +122,18 @@ public InetSocketAddress getSocketAddress() {
         return socketAddress;
     }
 
+    /**
+     * The HostSupplier may not have the ability to return the port
+     * and therefore the hashCode() may fail if the user provides his
```

Comment:
Also consider adding a note about this in the class javadoc, after the mention of a Host being immutable except for the status